src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/AddImports/IAddImportsService.cs

```diff
@@ -26,39 +26,37 @@ internal record struct AddImportPlacementOptions(
         [property: DataMember(Order = 1)] bool PlaceImportsInsideNamespaces,
         [property: DataMember(Order = 2)] bool AllowInHiddenRegions)
     {
-#if !CODE_STYLE
         public static async Task<AddImportPlacementOptions> FromDocumentAsync(Document document, CancellationToken cancellationToken)
-            => FromDocument(document, await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false));
-
-        public static AddImportPlacementOptions FromDocument(Document document, OptionSet documentOptions)
         {
-            var service = document.GetRequiredLanguageService<IAddImportsService>();
-
-            return new(
-                PlaceSystemNamespaceFirst: documentOptions.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, document.Project.Language),
-                PlaceImportsInsideNamespaces: service.PlaceImportsInsideNamespaces(documentOptions),
-                AllowInHiddenRegions: CanAddImportsInHiddenRegions(document));
+#if CODE_STYLE
+            var options = document.Project.AnalyzerOptions.GetAnalyzerOptionSet(await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false), cancellationToken);
+#else
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+#endif
+            return FromDocument(document, options);
         }
 
         private static bool CanAddImportsInHiddenRegions(Document document)
         {
+#if CODE_STYLE
+            return false;
```

Comment:
Why is this required? Can we instead move `ISpanMappingService` to shared layer and do one of the following?

1. Move the implementations of `ISpanMappingService` also down to the shared layer
2. Have no implementation of `ISpanMappingService` in the CodeStyle layer, so `GetService` call below returns `null`.
3. Have a default implementation of `ISpanMappingService` in the CodeStyle layer that returns defaults

Reply:
@mavasani I tried to do that, but it seems it will need lots of changes. Currently document services don't work in shared layer. `TextDocument` has an `IDocumentServiceProvider Services` property, but it's internal. I don't think we'll want to change public API in this PR.

There could be another way for doing that without affecting public API, but I can't figure it out.

Reply:
Sounds good, thanks